### DISCUSSION TITLE
Availibility on mobile devices

### DISCUSSION THREAD

**Discussion by freddidierRTE** [opfab/operatorfabric-core#3098] [Ideas]
Today the ui can be used on a mobile phone but the notification mechanism is not useable , we need to implement the use of standard  notification mechanism of mobiles devices.

An effort has to be done on the ui to have it fully compatible with mobiles device screen size

**Comment by Jeltel** [comment]
Is it possible to make Operator Fabric into a [Progressive Web Application](https://en.wikipedia.org/wiki/Progressive_web_application)? So that you can install Operator Fabric as an application, which also uses the native OS notification system.

**Reply by freddidierRTE** [reply]
yes that's i guess the solution

**Comment by freddidierRTE** [comment]
To be available on mobile , we cannot use actual long polling mechanism as it does not work when we exit opfab . For mobile we need to send notifications even if application is closesd.

One way of doing it is using push notification  : https://web.dev/push-notifications-overview/

This is a normalize solution : https://www.w3.org/TR/push-api/

@quinarygio  make a first test of push notification : see branch : https://github.com/opfab/operatorfabric-core/tree/push-notifications 

To be noted : Support on Safari (IOS) for web notification is really recent .
